python/cudf/cudf/core/frame.py

```diff
@@ -1324,10 +1165,18 @@ def fillna(
             else:
                 filled_data[col_name] = col.copy(deep=True)
 
-        return self._mimic_inplace(
-            self._from_data(data=filled_data, index=self._index),
-            inplace=inplace,
+        # TODO: This logic needs to move into the IndexedFrame class.
+        old_index = self._index
```

Comment:
Is this not gonna fail if you call the method from index? I think it should be guarded in the isinstance check below.

Reply:
Yes, thanks for reminding me. I addressed the TODOs and moved the index replacement into an IndexedFrame override, which should fix this problem.